### PR TITLE
Update schema for new deployment propreties.

### LINKED ISSUE #6063: Follow up work for allowing changes to container apps via CDK
Follow up work from https://github.com/dotnet/aspire/pull/5470

1. - [ ] - Update the schema to support deployment nodes for container.v1 and project.v1 resources (#6333)
2. - [x] - Fix secret handling (https://github.com/dotnet/aspire/pull/6005)

### PR DESCRIPTION
**Pull request by mitchdenny**
## Description

Moved `azure.bicep.v0` into definitions and referenced it from `project.v1` and `container.v1` as optional properties. Also referenced `azure.bicep.v0` in the `oneOf` clause and added test to make sure we didn't break by moving it to definitions.

Fixes #6063

## Checklist

- Is this feature complete?
  - [x] Yes. Ready to ship.
  - [ ] No. Follow-up changes expected.
- Are you including unit tests for the changes and scenario tests if relevant?
  - [x] Yes
  - [ ] No
- Did you add public API?
  - [ ] Yes
    - If yes, did you have an API Review for it?
      - [ ] Yes
      - [ ] No
    - Did you add `<remarks />` and `<code />` elements on your triple slash comments?
      - [ ] Yes
      - [ ] No
  - [x] No
- Does the change make any security assumptions or guarantees?
  - [ ] Yes
    - If yes, have you done a threat model and had a security review?
      - [ ] Yes
      - [ ] No
  - [x] No
- Does the change require an update in our Aspire docs?
  - [ ] Yes
    - Link to aspire-docs issue: 
  - [x] No

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/6333)

### CODE REVIEW

**Review by davidfowl — APPROVED** [review]

File: src/Schema/aspire-8.0.json [hunk 1]
```diff
@@ -20,7 +20,7 @@
           {
```
Comment (eerhardt) [inline src/Schema/aspire-8.0.json:20]: Is it weird that this file is still named `aspire-8.0`?

File: tests/Aspire.Hosting.Tests/Aspire.Hosting.Tests.csproj [hunk 2]
```diff
@@ -11,6 +11,7 @@
 
   <ItemGroup>
     <ProjectReference Include="..\..\src\Aspire.Hosting.AppHost\Aspire.Hosting.AppHost.csproj" IsAspireProjectResource="false" />
+    <ProjectReference Include="..\..\src\Aspire.Hosting.Azure.AppContainers\Aspire.Hosting.Azure.AppContainers.csproj" IsAspireProjectResource="false" />
```
Comment (eerhardt) [inline tests/Aspire.Hosting.Tests/Aspire.Hosting.Tests.csproj:14]: We've been trying to keep Azure specific tests out of the Aspire.Hosting.Tests. If/when we want to push the Azure specific libraries to an Azure owned repo, we would need to do work to split these new tests out.

Comment (eerhardt) [reply tests/Aspire.Hosting.Tests/Aspire.Hosting.Tests.csproj:14]: These new tests should be moved to

https://github.com/dotnet/aspire/blob/b87cab2d9a8bb06aa0c8e5795a11d4aa2789c9ec/tests/Aspire.Hosting.Azure.Tests/AzureSchemaTests.cs#L9-L27

Comment (davidfowl) [reply tests/Aspire.Hosting.Tests/Aspire.Hosting.Tests.csproj:14]: The JSON schema is a bit of a monolith... This is a reflection of our product at the moment.

Comment (eerhardt) [reply tests/Aspire.Hosting.Tests/Aspire.Hosting.Tests.csproj:14]: The JSON schema can be a monolith - it doesn't mean our tests need to be.

**Pull request merged**